### PR TITLE
SLING-5811 - Properly handle actual Resource's in Sightly data-sly-resource

### PR DESCRIPTION
**Pull request by vladbailescu**
- Added handling for actual Resource objects in data-sly-resource
- Added tests

### CODE REVIEW

File: bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java [hunk 1]
```diff
@@ -229,16 +238,24 @@ private Object getAndRemoveOption(Map<String, Object> options, String property)
 
     private void includeResource(final Bindings bindings, PrintWriter out, String script, String dispatcherOptions, String resourceType) {
         if (StringUtils.isEmpty(script)) {
-            LOG.error("Script path cannot be empty");
+            LOG.error("Resource path cannot be empty");
```
Comment (kwin) [inline bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java]: Can we rather throw an exception instead? That will also help with the new maven plugin.

File: bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java [hunk 2]
```diff
@@ -229,16 +238,24 @@ private Object getAndRemoveOption(Map<String, Object> options, String property)
...
             if (includeRes instanceof NonExistingResource || includeRes.isResourceType(Resource.RESOURCE_TYPE_NON_EXISTING)) {
                 includeRes = new SyntheticResource(request.getResourceResolver(), script, resourceType);
             }
+            includeResource(bindings, out, includeRes, dispatcherOptions, resourceType);
+        }
+    }
+
+    private void includeResource(final Bindings bindings, PrintWriter out, Resource includeRes, String dispatcherOptions, String resourceType) {
+        if (includeRes == null) {
+            LOG.error("Resource cannot be null");
```
Comment (kwin) [inline bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java]: I would prefer throwing an exception here as well. That way programming mistakes are detected earlier.

File: bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java (hunk 1 above)
Comment (vladbailescu) [inline bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java]: Makes sense and it's inline with https://github.com/apache/sling/blob/trunk/bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/IncludeRuntimeExtension.java#L95

File: bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java (hunk 2 above)
Comment (vladbailescu) [inline bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java]: Sure

File: bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java (hunk 1 above)
Comment (raducotescu) [inline bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java]: > That will also help with the new maven plugin.

@kwin: not really, as the plugin only receives the compilation warnings and errors. However I'd like to aim for consistency, so an exception makes more sense here.

**Pull request closed without merging**